mbuild/lib/recipes/polymer.py

```diff
@@ -182,25 +202,30 @@ def build(self, n, sequence="A", add_hydrogens=True):
 
         head_tail = [head_port, tail_port]
 
-        for i, compound in enumerate(self._end_groups):
-            if compound is not None:
-                if self._headtail[i] is not None:
-                    head_tail[i].update_separation(self._headtail[i])
-                self.add(compound)
-                force_overlap(compound, compound.labels["up"], head_tail[i])
-            else:
-                if add_hydrogens:
-                    hydrogen = H()
-                    # Defaut to 1/2 H-C bond len
-                    head_tail[i].update_separation(0.0547)
-                    hydrogen["up"].update_separation(0.0547)
-                    self.add(hydrogen)
-                    force_overlap(hydrogen, hydrogen["up"], head_tail[i])
+        if periodic_axis is not None:
+            self.create_periodic_bond(head_port, tail_port, axis=periodic_axis)
```

Comment:
Change to `self.create_periodic_bond(axis=periodic_axis)`

I don't think we need to have ports as input parameters here. The periodic bond will always be between head and tail, so we can use those properties in `create_periodic_bond()`